include/mgp.hpp

```diff
@@ -1220,104 +1220,98 @@ class Value {
   explicit Value(Path &&path) {
     ptr_ = mgp::value_make_path(path.ptr_);
-    delete &path;
     path.ptr_ = nullptr;
   }
 
   // Temporal type constructors:
 
   /// @brief Constructs a Date value from the copy of the given `date`.
-  explicit Value(const Date &date) : ptr_(mgp::value_make_date(date.ptr_)) {}
+  explicit Value(const Date &date) : ptr_(mgp::value_make_date(mgp::date_copy(date.ptr_, memory))) {}
   /// @brief Constructs a Date value and takes ownership of the given `path`.
   /// @note The behavior of accessing `date` after performing this operation is undefined.
   explicit Value(Date &&date) {
     ptr_ = mgp::value_make_date(date.ptr_);
-    delete &date;
     date.ptr_ = nullptr;
   }
 
   /// @brief Constructs a LocalTime value from the copy of the given `local_time`.
-  explicit Value(const LocalTime &local_time) : ptr_(mgp::value_make_local_time(local_time.ptr_)) {}
+  explicit Value(const LocalTime &local_time)
+      : ptr_(mgp::value_make_local_time(mgp::local_time_copy(local_time.ptr_, memory))) {}
   /// @brief Constructs a LocalTime value and takes ownership of the given `local_time`.
   /// @note The behavior of accessing `local_time` after performing this operation is undefined.
   explicit Value(LocalTime &&local_time) {
     ptr_ = mgp::value_make_local_time(local_time.ptr_);
-    delete &local_time;
     local_time.ptr_ = nullptr;
   }
 
   /// @brief Constructs a LocalDateTime value from the copy of the given `local_date_time`.
-  explicit Value(const LocalDateTime &local_date_time) : ptr_(mgp::value_make_local_date_time(local_date_time.ptr_)) {}
-
+  explicit Value(const LocalDateTime &local_date_time)
+      : ptr_(mgp::value_make_local_date_time(mgp::local_date_time_copy(local_date_time.ptr_, memory))) {}
   /// @brief Constructs a LocalDateTime value and takes ownership of the given `local_date_time`.
   /// @note The behavior of accessing `local_date_time` after performing this operation is undefined.
   explicit Value(LocalDateTime &&local_date_time) {
     ptr_ = mgp::value_make_local_date_time(local_date_time.ptr_);
-    delete &local_date_time;
     local_date_time.ptr_ = nullptr;
   }
 
   /// @brief Constructs a Duration value from the copy of the given `duration`.
-  explicit Value(const Duration &duration) : ptr_(mgp::value_make_duration(duration.ptr_)) {}
+  explicit Value(const Duration &duration)
+      : ptr_(mgp::value_make_duration(mgp::duration_copy(duration.ptr_, memory))) {}
   /// @brief Constructs a Duration value and takes ownership of the given `duration`.
   /// @note The behavior of accessing `duration` after performing this operation is undefined.
   explicit Value(Duration &&duration) {
     ptr_ = mgp::value_make_duration(duration.ptr_);
-    delete &duration;
     duration.ptr_ = nullptr;
   }
 
   Value(const Value &other) : Value(other.ptr_) {}
+  // Value(const Value &other) : Value(mgp::value_copy(other.ptr_, memory)) {}
```

Comment:
Remove commented part

Reply:
Thanks! I’ll run unit tests and remove the comments when everything’s working.